app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -105,6 +107,14 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         val autocompleteFilter = UrlAutoCompleteFilter()
         autocompleteFilter.load(context.applicationContext)
         setOnFilterListener { searchText, view -> autocompleteFilter.onFilter(searchText, view) }
+
+        setOnUserInputListener { hasUserChangedURLSinceEditTextFocused = true }
```

Comment:
Sorry this is confusing: it's another Kotlin rule that stacks with what I described before.

If your argument is an interface with a single method, Kotlin lets you pass in an anonymous function instead (since it can unambiguously determine what needs to be overridden). Then, going back to what I told you before (and as you mention above), Kotlin lets you pass in the anonymous function as the last argument outside the parentheses.

The reason Kotlin does this is because it greatly improves readability: single method interfaces are verbose to declare, they're redundant (`setOnUserInputListener` tells me what I need to know and both the interface name `OnUserInputListener` and the method `.onUserInput()` is redundant to that), and, when thinking about interop, there are a lot of these in Java (e.g. every callback). It's especially bad if you nest these.

In Java:
```java
setOnUserInputListener(new OnUserInputListener {
    @Override
    void onUserInput() {
        hasUserChangedURLSinceEditTextFocused = true;
    }
});
```
> you have to figure out "what interface is being filled again?"

If the method is appropriately named (as it is in this case), the actual interface generally doesn't matter.